src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentRuleBaseline/ManagedDatabase/SetAzureSqlManagedDatabaseVulnerabilityAssessmentRuleBaseline.cs

```diff
@@ -21,14 +21,13 @@ namespace Microsoft.Azure.Commands.Sql.VulnerabilityAssessment.Cmdlet.Vulnerabil
     /// <summary>
     /// Sets the Vulnerability Assessment baseline properties for a specific rule.
     /// </summary>
-    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlInstanceDatabaseVulnerabilityAssessmentRuleBaseline", SupportsShouldProcess = true, DefaultParameterSetName = SetByNameParameterSet), OutputType(typeof(ManagedDatabaseVulnerabilityAssessmentRuleBaselineModel))]
+    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlInstanceDatabaseVulnerabilityAssessmentRuleBaseline", SupportsShouldProcess = true), OutputType(typeof(ManagedDatabaseVulnerabilityAssessmentRuleBaselineModel))]
     public class SetAzureSqlManagedDatabaseVulnerabilityAssessmentRuleBaseline : SetAzureSqlVulnerabilityAssessmentRuleBaselineBase
     {
         /// <summary>
         /// Gets or sets the name of the managed instance to use.
         /// </summary>
-        [Parameter(ParameterSetName = SetByNameParameterSet,
-            Mandatory = true,
+        [Parameter(Mandatory = true,
```

Comment:
Why this parameter set was removed?

Reply:
I discovered that our piping scenario (in set baseline) was broken. When this parameter set was defined, powershell didn't recognize the InputObject and kept trying to serialize it into key-value pairs. This broke the pipeline scenario completely.
With this parameter removed, powershell recognizes only one parameter set with InputObject an optional parameter, and is able to recognize it as an object (and piping works perfectly). This doesn't change the cmdlet's behavior, just fixes the piping scenario. I checked it both manually and ran all the tests again

Reply:
We need to investigate if this would reflect in a real breaking change or not. By removing the `SetByNameParameterSet`, if a script were to be using it before, that is, without specifying the `-InputObject` or piping any object. Will this continue to work as before? That is, will the code not try to use the `InputObject` parameter but instead use the other information `InstanceName` and `DatabaseName` ?

Reply:
I have checked both parameter sets manually. The cmdlet accepts InstanceName and DatabaseName (with no InputObject). Also the tests check this scenario and I have re-run them and they all passed

Reply:
@talhers I just realized that `-InputObject` is not mandatory and I believe that is why your piping scenarios are failing. Could you please make `-InputObject` mandatory. Revert back the changes you made that removed the `SetByNameParameterSet` and test if piping works now?

Reply:
That's the first thing I did :)
making InputObject mandatory (and restoring SetByNameParameterSet) doesn't solve the problem, powershell still attempts to serialize it. This only results in more breaking changes and doesn't solve the problem.

I tested all scenarios both in the tests and manually, and there is no change to existing cmdlet behavior. This only fixes the piping bug.

If everything else is ok, can you merge my PR? Many thanks!

Reply:
Having InputObject in the same parameter set with the identity parameters is not allowed.  Please come to office hours, or set up some time tto work with us on resolving the underlying issue.